executor/join.go

```diff
@@ -593,7 +958,10 @@ func (e *HashJoinExec) join2Chunk(workerID uint, probeSideChk *chunk.Chunk, hCtx
 		return false, joinResult
 	}
 
-	hCtx.initHash(probeSideChk.NumRows())
+	numRows := probeSideChk.NumRows()
+	hCtx.initHash(numRows)
+	// By now, path 1 and 2 won't be conducted at the same time.
```

Comment:
Then why not use `if isNAAJ` to make sure path 1 and path 2 could not be conducted at the same time?

Reply:
cause we guarantee the EQcondition and NAEQcondition couldn't be filled at same time before.